src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.cs

```diff
@@ -296,6 +296,13 @@ internal bool IsSupportedDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diag
                 return false;
             }
 
+            if (analyzer is ArtifactProducerDiagnosticAnalyzer)
+            {
+                // artifact generators only run if we're generating output files to disk (and thus have an artifact
+                // stream). Otherwise they are suppressed as there's no purpose running them.
+                return analyzerExecutor.CreateArtifactStream == null;
```

Comment:
I'm not sure i understand the question.  This is definitely important.  If this is the analyzer created to run artifact production, we want to consider that analyzer suppressed when the compiler is not in 'artifact production' mode.  There's no point running these otherwise as:
1.they can be quite expensive
2. they are run in contexts when trying creat such a stream isn't supported.